src/FakeItEasy/Core/MethodInfoManager.cs

```diff
@@ -88,7 +88,7 @@ where method.HasSameBaseMethodAs(methodTargetPair.TargetMethod)
         {
             var baseDefinition = method.GetBaseDefinition();
 
-            if (!baseDefinition.DeclaringType.IsInterface || !TypeImplementsInterface(type, baseDefinition.DeclaringType))
+            if (!baseDefinition.DeclaringType!.IsInterface || !TypeImplementsInterface(type, baseDefinition.DeclaringType))

```

Comment:
DeclaringType can only be null for non-nested types, module-level methods, or dynamic methods.